### PR TITLE
Correcting the terraform apply retry logic

### PR DESCRIPTION
**Pull request by Rajalakshmi-Girish**
[#1354 ](https://github.ibm.com/powercloud/container-dev/issues/1354)
kubetest2-plugin failed to rerun the tf when tf apply failed first time for the following reason:
`For` loop that does the retry has a `break` command to stop the iteration when tf apply is successful.
This break of loop is happening at the first iteration itself (irrespective of tf apply being successful or not) and **thus the flow is exiting the `For` loop and there by having no retry.**
Hence placed the print of tf state and `break` command in `else` block so that it occurs only when there is tf apply success i.e when err= nil for tf apply.

```
+               } else {
+                       fmt.Printf("terraform state at: %s\n", path)
+                       break
                }
-               fmt.Printf("terraform state at: %s\n", path)
-               break
        }
```

### CODE REVIEW

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/kubetest2-plugins/pull/25#" title="Author self-approved">Rajalakshmi-Girish</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **mkumatag** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @mkumatag` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Fkubetest2-plugins).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/ppc64le-cloud/kubetest2-plugins/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: kubetest2-tf/deployer/deployer.go
Comment (mkumatag) [inline kubetest2-tf/deployer/deployer.go:146]: oerr is not printed in this block., otherwise looks okay

**Review by mkumatag — CHANGES_REQUESTED** [review]

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/kubetest2-plugins/pull/25#issuecomment-742243978" title="Approved">mkumatag</a>*, *<a href="https://github.com/ppc64le-cloud/kubetest2-plugins/pull/25#" title="Author self-approved">Rajalakshmi-Girish</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Fkubetest2-plugins).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ppc64le-cloud/kubetest2-plugins/blob/master/OWNERS)~~ [mkumatag]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
